### ISSUE TITLE
Please split up this repository

### ISSUE THREAD

**Issue by tarsius** [zenspider/elisp#5]
Could you please consider splitting up this repository. It contains libraries that are in no way related to one another and that is inconvenient for users who are only interested in some of these libraries.

Most users of one of these libraries install it by installing a package from Melpa. For those users it does not matter whether all the libraries live in one repository or are neatly divided into dedicated repositories. So it's not unreasonable to reject this request and the basis that this is more convenient for yourself.

However I would still like to encourage you to do it anyway, for the benefit of the potentially most "useful" users - those that contribute to one of your libraries. Here's an example of why having unrelated things in one repository can cause problems: Imagine someone has installed several of your unrelated libraries as Melpa packages, and then decides to contribute a complex change to one of them. So she clones the repository, adds it to the `load-path` and starts hacking. Until she is done and you have merged the changes, the other libraries - including the ones she does not care for - are also sitting on the `load-path`. Meanwhile you fix some crucial  bugs in to one of the other libraries which she happens to have installed using `package.el`. Unfortunately though, she is sitting on a feature branch and doesn't pull your `master` and her checkout happens to be earlier on the `load-path` than the packages she has installed using the package manager. So the bug isn't fixed for here.

Sure she could change the order of the `load-path` or put only certain libraries from the checkout on the `load-path` by using symlinks. Or regularly rebase her feature branch. She could probably do quite a few other simple things to work around this.

My point however is that she shouldn't have to do it. Unrelated things do not belong in the same repository, at least not when that repository is publicly available. Doing so might be inconvenient at first but I think that doing so will benefit you in the long run. For example you will get a much more meaningful history.

**Comment by tarsius** [comment]
I am once more trying to mirror all the packages available from Melpa on the Emacsmirror two, and would therefore like to ask you to create separate repositories for at least those libraries that are available as individual packages on Melpa.

As far as I can tell, these are `autotest` and `toggle`.

**Comment by zenspider** [comment]
If melpa can handle packaging subsets of a repo, shouldn't your mirror? I keep my melpa packages up to date and working (and hopefully tagged for the stables). I don't really want to jump through hoops and make my development process more difficult for the benefit of a mirror I don't even use.

**Comment by tarsius** [comment]
> If melpa can handle packaging subsets of a repo, shouldn't your mirror?

Melpa is a package archive, the Emacsmirror is a mirror of repository which ideally contain a single package, or multiple closely related packages. So, no.

> I don't really want to jump through hoops and make my development process more difficult for the benefit of a mirror I don't even use.

Fair enough. I won't mirror those packages then. I'll write a blog post about using Git submodules to install and maintain Emacs packages in a few weeks. Hopefully you'll change your mind then.

**Comment by tarsius** [comment]
Not keeping unrelated packages inside the same repository allows users to install one of those package as a git submodule without also installing all the others. I explain why installing packages as submodules enables users to more easily contribute to the packages that they use in [Assimilate Emacs packages using Git submodules](https://emacsair.me/2016/05/17/assimilate-emacs-packages-as-git-submodules).

I mentioned the Melpa package because I maintain the Emacsmirror, a mirror of Emacs package _repositories_, and have been trying to mirror all packages that are available from Melpa. Something that I have accomplished some time ago.

I used to not mirror any packages whose repositories bundle third-party packages. I mirror them anyway now because I want to mirror everything that is available from Melpa, but that requires a lot of work to fixup the extracted metadata. And that is all I can do, fixup the metadata, if a user actually adds a "broken" repository to the `load-path`, then issues like the below will arise.

When a repository bundles third-party libraries, then it is undefined which version will get loaded, the bundled version or the authoritative version. That's an issue because bundled versions in experience tend to not get updated for years. The user, being unaware that s/he is using an outdated version, might then report bugs against the upstream repository, and for obvious reasons upstream won't be able to reproduce them.
